tenacity/__init__.py

```diff
@@ -196,52 +191,57 @@ def _first_set(first, second):
 class RetryError(Exception):
     """Encapsulates the last attempt instance right before giving up."""
 
-    def __init__(self, last_attempt):
+    def __init__(self, last_attempt: "Future") -> None:
         self.last_attempt = last_attempt
-        super(RetryError, self).__init__(last_attempt)
+        super().__init__(last_attempt)
 
-    def reraise(self):
+    def reraise(self) -> t.NoReturn:
         if self.last_attempt.failed:
             raise self.last_attempt.result()
         raise self
 
-    def __str__(self):
+    def __str__(self) -> str:
         return "{0}[{1}]".format(self.__class__.__name__, self.last_attempt)
 
 
-class AttemptManager(object):
+class AttemptManager:
     """Manage attempt context."""
 
-    def __init__(self, retry_state):
+    def __init__(self, retry_state: "RetryCallState"):
         self.retry_state = retry_state
 
     def __enter__(self):
         pass
 
-    def __exit__(self, exc_type, exc_value, traceback):
+    def __exit__(
+            self,  # noqa:BLK100
```

Comment:
modern `black` will format this style (maybe we need to add it on CI?)

Reply:
We have `flake8-black` in the CI if that's what you mean.

Reply:
`black --check --diff .` is more consistent. And I suggest to set project line length:
`.editorconfig` file:
```toml
[*.{py,pyx,pxd,pyi}]
insert_final_newline = true
trim_trailing_whitespace = true
indent_style = 'space'
indent_size = 4
max_line_length = 120
```
and `pyproject.toml`:
```toml
[tool.black]
target-version = ["py36", "py37", "py38", "py39"]
safe = true
line-length = 120
```
(120 is mine normal number, it can be any, config validated on `black==21.6b0`)

Reply:
That should be done if anything in another PR.

Last time I checked, setting black line-length was not good idea because it'd be conflict with e.g. flake8, but if you can make everything work together I'm fine.

Reply:
Yes, it will be separate PR.